### PR TITLE
docs: Add known limitations callouts to Max Client Disconnect section

### PR DESCRIPTION
**Pull request by DerekStrickland**

### CODE REVIEW

File: website/content/docs/job-specification/group.mdx [hunk 1]
```diff
@@ -329,6 +329,46 @@ group "second" {
 }
 ```
 
+#### Known Limitations
+
+The `max_client_disconnect` configuration is a new operating model for Nomad. Due
+to the nature of the change and the challenges it does and does not address, there
+are some known limitations to the feature that are detailed below.
```
Comment (tgross) [inline website/content/docs/job-specification/group.mdx:336]: This language becomes stale almost immediately, and at least the first two behaviors we're describing aren't going to change. I'd just leave this section and the top header out.

File: website/content/docs/job-specification/group.mdx [hunk 2]
```diff
@@ -329,6 +329,46 @@ group "second" {
...
+
+Similar to servers, clients running older versions of Nomad are unaware of the
+new node and allocation statuses. Unlike servers, there is currently no guard
+logic that checks all client versions during the scheduling process. The performance
+impact of such a check during scheduling is prohibitive. Attempting to leverage
+the `max_client_disconnect` feature in clusters with mixed client versions is _not
+supported_. Operators must ensure all clients meet the 1.3.0 minimum version requirement
+prior to deploying jobs with this configuration. If a job with this configuration
+is deployed to a cluster that does not meet this requirement, operators should
+run `nomad job stop -purge` on the job and remove the configuration before re-deploying.
```
Comment (tgross) [inline website/content/docs/job-specification/group.mdx:359]: Most of this kind of stuff applies to _every_ new feature, and it's very wordy, and none of it is going to change over time. We have a warning block syntax (`~> **Note:** `), so I'd just wrap this all up like so:

```suggestion
~> **Note:** The `max_client_disconnect` feature is only supported on Nomad 
version 1.3.0 and above. If you run a job with `max_client_disconnect` on servers 
where some servers are not upgraded to 1.3.0, the `max_client_disconnect` 
flag will be _ignored_. Deploying a job with `max_client_disconnect` to a 
`datacenter` of Nomad clients where all clients are not 1.3.0 or above is unsupported.
```

File: website/content/docs/job-specification/group.mdx [hunk 3]
```diff
@@ -329,6 +329,46 @@ group "second" {
...
+##### Consul, Vault, and Consul Template
+
+Jobs that rely on network services to run or deploy will fail if there is no
+connection to the network service. For example, if the job includes a template
+that uses the new `nomadService` function for Nomad Native Service Discovery,
+and the client cannot reach the Nomad server, the task will fail when it next
+attempts to render. Similarly, jobs that rely on Consul or Vault will be affected
+by connectivity loss. These jobs will adhere to the client's retry configuration,
+but if they are ultimately unable to reach Consul or Vault after their reaching
+their specified retry threshold, they will fail as they do previous to 1.3.0.
```
Comment (tgross) [inline website/content/docs/job-specification/group.mdx:370]: Is this new information specific to this feature? Doesn't this already happen to all template tasks?

Comment (DerekStrickland) [reply website/content/docs/job-specification/group.mdx:347]: Good point. I'll delete this.

**Review by tgross — APPROVED** [review]

**Pull request merged**